testutil/daemon/daemon.go

```diff
@@ -140,13 +135,41 @@ func New(t testingT, ops ...func(*Daemon)) *Daemon {
 		dockerdBinary:   defaultDockerdBinary,
 		swarmListenAddr: defaultSwarmListenAddr,
 		SwarmPort:       DefaultSwarmPort,
-		log:             t,
+		log:             nopLog{},
 	}
 
 	for _, op := range ops {
 		op(d)
 	}
 
+	return d, nil
+}
+
+// New returns a Daemon instance to be used for testing.
+// This will create a directory such as d123456789 in the folder specified by
+// $DOCKER_INTEGRATION_DAEMON_DEST or $DEST.
+// The daemon will not automatically start.
+func New(t testingT, ops ...func(*Daemon)) *Daemon {
+	if ht, ok := t.(testutil.HelperT); ok {
+		ht.Helper()
+	}
+	dest := os.Getenv("DOCKER_INTEGRATION_DAEMON_DEST")
+	if dest == "" {
+		dest = os.Getenv("DEST")
+	}
+
+	switch v := t.(type) {
+	case namer:
+		dest = filepath.Join(dest, v.Name())
+	case testNamer:
+		dest = filepath.Join(dest, v.TestName())
+	}
+	assert.Check(t, dest != "", "Please set the DOCKER_INTEGRATION_DAEMON_DEST or the DEST environment variable")
+
+	t.Logf("Creating a new daemon")
```

Comment:
not sure if branches got messed up in the refactor, but this is missing the dest information compared to https://github.com/moby/moby/pull/39943/files#diff-c81dbe4e6a808b9708f11bbf96deb8fdL111